### DISCUSSION TITLE
Unable to resetStore while using Nuxt

### DISCUSSION THREAD

**Discussion by gijo-varghese** [Akryum/vue-apollo#283] [Q&A]
I would like to reset all queries when web socket is reconnected. Here is what I tried. But it's not working

```
 link.subscriptionClient.onReconnected(() => {
      console.log("Reconnected...");
      this.$apollo.provider.defaultClient.resetStore(); // not working
      $nuxt.$apollo.provider.defaultClient.resetStore(); // not working
    });
```

When I tried `console.log($nuxt.$apollo);` I got:
![screenshot from 2018-05-17 19-30-09](https://user-images.githubusercontent.com/23092846/40182350-d5897a34-5a08-11e8-9c91-f9972344d7bc.png)

**Comment by gijo-varghese** [comment]
Full code:
```
import { HttpLink } from "apollo-link-http";
import { InMemoryCache } from "apollo-cache-inmemory";
import { WebSocketLink } from "apollo-link-ws";
import { SubscriptionClient } from "subscriptions-transport-ws";

export default () => {
  let link = null;
  if (process.server) {
    link = new HttpLink({
      uri: process.env.POSTMASTER_GRAPHQL_URL + "/graphql"
    });
  } else {
    const client = new SubscriptionClient(
      process.env.POSTMASTER_SUBSCRIPTIONS_URL + "/subscriptions",
      {
        reconnect: true,
        lazy: true
      }
    );

    link = new WebSocketLink(client);

    link.subscriptionClient.maxConnectTimeGenerator.duration = () =>
      link.subscriptionClient.maxConnectTimeGenerator.max;

    link.subscriptionClient.onReconnected(() => {
      // $nuxt.$apollo.provider.defaultClient.resetStore();
    });
  }
  const cache = new InMemoryCache();
  return {
    link,
    cache
  };
};
```

**Comment by Akryum** [comment]
Maybe this is an issue with apollo-client?

**Comment by gijo-varghese** [comment]
Ok. So should I post this in their repo?

**Comment by ScreamZ** [comment]
@gijo-varghese Any fix found ? Not sure this is working for me too

**Comment by gijo-varghese** [comment]
Working or not working? I haven't found a fix yet

**Comment by ScreamZ** [comment]
@gijo-varghese Ping me if you're having more informations. I'll try investigate inside the library directly.

**Comment by gijo-varghese** [comment]
@Akryum I've updated the `@nuxt/apollo`, so `vue-apollo` is now "version": "3.0.0-beta.14". But I'm facing the same error.  Both `client` and `provider` is `null` as before

**Comment by Akryum** [comment]
I think you can put the socket listeners in a Nuxt middleware instead?

**Comment by Akryum** [comment]
I'm not familiar with Nuxt, but it seems you need to put it where you will have access to the ApolloClient (and only on client-side).
